contracts/BaseStrategy.sol

```diff
@@ -228,6 +236,16 @@ abstract contract BaseStrategy {
         require(msg.sender == keeper || msg.sender == strategist || msg.sender == governance(), "!authorized");
         _;
     }
+    modifier useChi() {
+        if (address(chiToken) == address(0)) {
+            _;
+            return;
+        }
+        uint256 gasStart = gasleft();
+        _;
+        uint256 gasSpent = 21000 + gasStart - gasleft() + 16 * msg.data.length;
+        chiToken.freeFromUpTo(msg.sender, (gasSpent + 14154) / 41130);
```

Comment:
Is there a way to check the balance of tokens available? or does `freeFromUpTo` automatically take this into account?

Reply:
Yes it does  automatically take this into account the number of token that can be used.